### PR TITLE
test(cli): add test for `add` and `create` commands

### PR DESCRIPTION
**Pull request by eunjae-lee**
## What?

This PR adds tests for `add` and `create` commands. `deploy` command will be dealt in a separate PR.

## Why?

It's a follow-up of #238 

## How to test? (optional)

```sh
yarn test:cli
```

### CODE REVIEW

**Comment by eunjae-lee** [comment]
Current dependencies on/for this PR:
* main
  * **PR #238** <a href="https://app.graphite.dev/github/pr/storyblok/field-plugin/238" target="_blank">Graphite</a> 
    * **PR #239** <a href="https://app.graphite.dev/github/pr/storyblok/field-plugin/239" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/storyblok/field-plugin/239?utm_source=stack-comment).

File: packages/cli/src/commands/create/__tests__/index.test.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import { vi } from 'vitest'
+import { create } from '../'
+import {
+  isValidPackageManager,
+  selectPackageManager,
+  selectRepositoryStructure,
+  isValidStructure,
+} from '../../../utils'
+
+vi.mock('../../../utils')
```
Comment (BibiSebi) [inline packages/cli/src/commands/create/__tests__/index.test.ts:10]: I might have too little knowledge on this one, but is there a reason why this file imports utils in three different ways? 

See line 8,10 and 16

**Review by BibiSebi — APPROVED** [review]
Huge thanks on creating these test cases. Well done! 👏

File: packages/cli/src/commands/create/__tests__/index.test.ts (hunk 1 above)
Comment (eunjae-lee) [reply packages/cli/src/commands/create/__tests__/index.test.ts:10]: Ah, let me explain.

By running `vi.mock('../../../utils')`, we mock everything that is exported from `utils`.

Among those mocked exports, we're importing these 4 functions:

```js
import {
  isValidPackageManager,
  selectPackageManager,
  selectRepositoryStructure,
  isValidStructure,
} from '../../../utils'
```

The mocked functions do nothing, but we are importing these functions to assign a mock implementation for testing purpose. For example, we can do things like:

```js
vi.mocked(isValidPackageManager).mockImplementation(() => true);

vi.mocked(isValidPackageManager).mockImplementation(() => false);

vi.mocked(isValidPackageManager).mockImplementationOnce(() => true);
```

Then... what about line 19?

```js
utils = await vi.importActual('../../../utils')
```

This actually imports the unmocked real version of utils module.
Although we mocked everything, we sometimes need the real implementation of a certain methods.

For example, in this test case, we're using all the mocked things from `utils` except for this `isValidPackageManager` method. I'm putting the actual implementation back.

https://github.com/storyblok/field-plugin/blob/2ba6f82528166dd0f2b7422627d2a9d011087036/packages/cli/src/commands/create/__tests__/index.test.ts#L26C1-L33

You can read more about this at https://vitest.dev/guide/mocking.html

Let me know if you have any question :)

**Comment by eunjae-lee** [comment]
@eunjae-lee started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/storyblok/field-plugin/239).

**Comment by eunjae-lee** [comment]
[Graphite](https://app.graphite.dev/github/pr/storyblok/field-plugin/239) couldn't merge this pull request because a downstack PR #238 failed to merge.

**Comment by eunjae-lee** [comment]
@eunjae-lee merged this pull request with [Graphite](https://app.graphite.dev/github/pr/storyblok/field-plugin/239).

**Pull request merged**